hook/registry.go

```diff
@@ -114,6 +122,11 @@ func (r *Registry) RegisterHook(name string, f func() error) {
 	})
 }
 
+// RegisterMetrics registers the metric that is reported by the charm.
+func (r *Registry) RegisterMetric(name string, kind charm.MetricType, description string) {
+	r.metrics.Metrics[name] = charm.Metric{Type: kind, Description: description}
```

Comment:
Perhaps this should panic if a metric is registered twice with the same name and different attributes? Something to consider for a subsequent PR.